#### import required libraries

In [211]:
import random # library for random number generation
import numpy as np # library for vectorized computation
import pandas as pd # library to process data as dataframes
import requests # 'requests' offers the most friendly API for opening files, including JSON support
from geopy.geocoders import Nominatim # convert an address into latitude and longitude values
import json # library to handle JSON files
from pandas import json_normalize 
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)
from sklearn.cluster import KMeans # import k-means from clustering stage
import folium # map rendering library

print('Libraries imported.')

Libraries imported.


#### import the BeautifulSoup library so we can parse HTML and XML documents

In [212]:
from bs4 import BeautifulSoup

# specify which URL/web page we are going to be scraping

url = "https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M"

# open the url and put the html into the soup variable

response = requests.get(url)
html = response.text
soup = BeautifulSoup(html,'html.parser')

#### load the values in each row into the fields A, B and C

In [213]:
all_table=soup.find('table', class_='wikitable sortable')

A=[]
B=[]
C=[]

for row in all_table.findAll('tr'):
    
    cells=row.findAll('td')
    
    if len(cells)==3:
        A.append(cells[0].find(text=True))
        B.append(cells[1].find(text=True))
        C.append(cells[2].find(text=True))

#### The dataframe will consist of three columns: PostalCode, Borough, and Neighbourhood

In [214]:
dfToronto = pd.DataFrame(A,columns=['PostalCode'])
dfToronto['Borough']=B
dfToronto['Neighbourhood']=C

# had to clean the data by removing \n that has appeared at the end of each text string in each column ... don't know why?!

dfToronto['PostalCode'] = dfToronto['PostalCode'].str.rstrip('\n')
dfToronto['Borough'] = dfToronto['Borough'].str.rstrip('\n')
dfToronto['Neighbourhood'] = dfToronto['Neighbourhood'].str.rstrip('\n')

#### Only process the cells that have an assigned Borough. Ignore cells with a Borough that is Not assigned

In [215]:
dfToronto['Borough'].replace('Not assigned', np.nan, inplace=True)
dfToronto.dropna(subset=['Borough'], inplace=True)
dfToronto.head()

PostalCode           Borough                                Neighbourhood
2        M3A        North York                                    Parkwoods
3        M4A        North York                             Victoria Village
4        M5A  Downtown Toronto                    Regent Park, Harbourfront
5        M6A        North York             Lawrence Manor, Lawrence Heights
6        M7A  Downtown Toronto  Queen's Park, Ontario Provincial Government

#### Combine rows so that each row has a unique postal code.  Seperate multiple dfTorOnly with a comma

In [216]:
dfToronto.groupby('PostalCode').agg({'Borough':'first','Neighbourhood': ', '.join}).reset_index()

PostalCode           Borough  \
0          M1B       Scarborough   
1          M1C       Scarborough   
2          M1E       Scarborough   
3          M1G       Scarborough   
4          M1H       Scarborough   
5          M1J       Scarborough   
6          M1K       Scarborough   
7          M1L       Scarborough   
8          M1M       Scarborough   
9          M1N       Scarborough   
10         M1P       Scarborough   
11         M1R       Scarborough   
12         M1S       Scarborough   
13         M1T       Scarborough   
14         M1V       Scarborough   
15         M1W       Scarborough   
16         M1X       Scarborough   
17         M2H        North York   
18         M2J        North York   
19         M2K        North York   
20         M2L        North York   
21         M2M        North York   
22         M2N        North York   
23         M2P        North York   
24         M2R        North York   
25         M3A        North York   
26         M3B        North York   
27         M3C        North York   
28         M3H        North York   
29         M3J        North York   
30         M3K        North York   
31         M3L        North York   
32         M3M        North York   
33         M3N        North York   
34         M4A        North York   
35         M4B         East York   
36         M4C         East York   
37         M4E      East Toronto   
38         M4G         East York   
39         M4H         East York   
40         M4J         East York   
41         M4K      East Toronto   
42         M4L      East Toronto   
43         M4M      East Toronto   
44         M4N   Central Toronto   
45         M4P   Central Toronto   
46         M4R   Central Toronto   
47         M4S   Central Toronto   
48         M4T   Central Toronto   
49         M4V   Central Toronto   
50         M4W  Downtown Toronto   
51         M4X  Downtown Toronto   
52         M4Y  Downtown Toronto   
53         M5A  Downtown Toronto   
54         M5B  Downtown Toronto   
55         M5C  Downtown Toronto   
56         M5E  Downtown Toronto   
57         M5G  Downtown Toronto   
58         M5H  Downtown Toronto   
59         M5J  Downtown Toronto   
60         M5K  Downtown Toronto   
61         M5L  Downtown Toronto   
62         M5M        North York   
63         M5N   Central Toronto   
64         M5P   Central Toronto   
65         M5R   Central Toronto   
66         M5S  Downtown Toronto   
67         M5T  Downtown Toronto   
68         M5V  Downtown Toronto   
69         M5W  Downtown Toronto   
70         M5X  Downtown Toronto   
71         M6A        North York   
72         M6B        North York   
73         M6C              York   
74         M6E              York   
75         M6G  Downtown Toronto   
76         M6H      West Toronto   
77         M6J      West Toronto   
78         M6K      West Toronto   
79         M6L        North York   
80         M6M              York   
81         M6N              York   
82         M6P      West Toronto   
83         M6R      West Toronto   
84         M6S      West Toronto   
85         M7A  Downtown Toronto   
86         M7R       Mississauga   
87         M7Y      East Toronto   
88         M8V         Etobicoke   
89         M8W         Etobicoke   
90         M8X         Etobicoke   
91         M8Y         Etobicoke   
92         M8Z         Etobicoke   
93         M9A         Etobicoke   
94         M9B         Etobicoke   
95         M9C         Etobicoke   
96         M9L        North York   
97         M9M        North York   
98         M9N              York   
99         M9P         Etobicoke   
100        M9R         Etobicoke   
101        M9V         Etobicoke   
102        M9W         Etobicoke   

                                         Neighbourhood  
0                                       Malvern, Rouge  
1               Rouge Hill, Port Union, Highland Creek  
2                    Guildwood, Morningside, West Hill  
3                              

#### If a cell has a Borough but a Not assigned Neighbourhood, then the Neighbourhood will be the same as the Borough

In [217]:
# print(df.loc[df['Neighbourhood'] == 'Not assigned']) ... this shows that there are no 'Not assigned' values in Neighbourhood

# But in case there were, this function replaces the Neighbourhood not assigned value with the bourough value

dfToronto['Neighbourhood']=dfToronto['Borough'].where(dfToronto['Neighbourhood'].eq('Not assigned'),dfToronto['Neighbourhood'])
dfToronto.head()


PostalCode           Borough                                Neighbourhood
2        M3A        North York                                    Parkwoods
3        M4A        North York                             Victoria Village
4        M5A  Downtown Toronto                    Regent Park, Harbourfront
5        M6A        North York             Lawrence Manor, Lawrence Heights
6        M7A  Downtown Toronto  Queen's Park, Ontario Provincial Government

#### In the last cell of your notebook, use the .shape method to print the number of rows of your dataframe

In [218]:
dfToronto.shape

(103, 3)

#### Load the latitude and longitude coordinates from the csv file downloaded to local machine

In [219]:
dfGeoloc = pd.read_csv('/Users/leostepan/AnacondaProjects/Geospatial_Coordinates.csv')

# rename the postal code field in dfGeoloc to match dfToronto and enable the merge

dfGeoloc.rename(columns={'Postal Code': 'PostalCode'}, inplace=True) 
dfGeoloc.head()

PostalCode   Latitude  Longitude
0        M1B  43.806686 -79.194353
1        M1C  43.784535 -79.160497
2        M1E  43.763573 -79.188711
3        M1G  43.770992 -79.216917
4        M1H  43.773136 -79.239476

#### Merge the two dataframes on the PostalCode field

In [220]:
df_merge = pd.merge(dfToronto, dfGeoloc, on='PostalCode')
df_merge.head(12)

PostalCode           Borough  \
0         M3A        North York   
1         M4A        North York   
2         M5A  Downtown Toronto   
3         M6A        North York   
4         M7A  Downtown Toronto   
5         M9A         Etobicoke   
6         M1B       Scarborough   
7         M3B        North York   
8         M4B         East York   
9         M5B  Downtown Toronto   
10        M6B        North York   
11        M9B         Etobicoke   

                                        Neighbourhood   Latitude  Longitude  
0                                           Parkwoods  43.753259 -79.329656  
1                                    Victoria Village  43.725882 -79.315572  
2                           Regent Park, Harbourfront  43.654260 -79.360636  
3                    Lawrence Manor, Lawrence Heights  43.718518 -79.464763  
4         Queen's Park, Ontario Provincial Government  43.662301 -79.389494  
5             Islington Avenue, Humber Valley Village  43.667856 -79.532242  
6                                      Malvern, Rouge  43.806686 -79.194353  
7                                           Don Mills  43.745906 -79.352188  
8                     Parkview Hill, Woodbine Gardens  43.706397 -79.309937  
9                            Garden District, Ryerson  43.657162 -79.378937  
10                                          Glencairn  43.709577 -79.445073  
11  West Deane Park, Princess Gardens, Martin Grov...  43.650943 -79.554724

#### Explore and cluster the dfTorOnly in Toronto, working only with Boroughs that contain the word Toronto

In [221]:
dfTorOnly = df_merge[df_merge.Borough.str.contains('Toronto',case=False)]
dfTorOnly.head(10)

PostalCode           Borough                                Neighbourhood  \
2         M5A  Downtown Toronto                    Regent Park, Harbourfront   
4         M7A  Downtown Toronto  Queen's Park, Ontario Provincial Government   
9         M5B  Downtown Toronto                     Garden District, Ryerson   
15        M5C  Downtown Toronto                               St. James Town   
19        M4E      East Toronto                                  The Beaches   
20        M5E  Downtown Toronto                                  Berczy Park   
24        M5G  Downtown Toronto                           Central Bay Street   
25        M6G  Downtown Toronto                                     Christie   
30        M5H  Downtown Toronto                     Richmond, Adelaide, King   
31        M6H      West Toronto                 Dufferin, Dovercourt Village   

     Latitude  Longitude  
2   43.654260 -79.360636  
4   43.662301 -79.389494  
9   43.657162 -79.378937  
15  43.651494 -79.375418  
19  43.676357 -79.293031  
20  43.644771 -79.373306  
24  43.657952 -79.387383  
25  43.669542 -79.422564  
30  43.650571 -79.384568  
31  43.669005 -79.442259

#### Restricting the Borough field to cells containnig Toronto reduces the size of the dataframe from 103 rows down to 39 rows

In [222]:
dfTorOnly.shape

(39, 5)

#### There are now four unique Boroughs to use for clustering

In [223]:
dfTorOnly.Borough.unique()

array(['Downtown Toronto', 'East Toronto', 'West Toronto',
       'Central Toronto'], dtype=object)

#### Create a map of Toronto for plotting

In [224]:
import folium # map rendering library

# !conda install -c conda-forge geopy --yes # uncomment this line if you haven't completed the Foursquare API lab
from geopy.geocoders import Nominatim # convert an address into latitude and longitude values

address = 'Toronto, ON'

geolocator = Nominatim(user_agent="to_explorer")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print('The geograpical coordinates of Toronto are {}, {}.'.format(latitude, longitude))

# create map of Toronto using latitude and longitude values
map_Toronto = folium.Map(location=[latitude, longitude], zoom_start=10)

# add markers to map
for lat, lng, Borough, Neighbourhood in zip(dfTorOnly['Latitude'], dfTorOnly['Longitude'], dfTorOnly['Borough'], dfTorOnly['Neighbourhood']):
    label = '{}, {}'.format(Neighbourhood, Borough)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_Toronto)  
    
map_Toronto

The geograpical coordinates of Toronto are 43.6534817, -79.3839347.


#### Slice the original dataframe to create seperate dataframes for each of the four boroughs that contain Toronto in their name

In [225]:
Downtown_data = dfTorOnly[dfTorOnly['Borough'] == 'Downtown Toronto'].reset_index(drop=True)
Central_data = dfTorOnly[dfTorOnly['Borough'] == 'Central Toronto'].reset_index(drop=True)
East_data = dfTorOnly[dfTorOnly['Borough'] == 'East Toronto'].reset_index(drop=True)
West_data = dfTorOnly[dfTorOnly['Borough'] == 'West Toronto'].reset_index(drop=True)

In [226]:
Central_data.head()

PostalCode          Borough  \
0        M4N  Central Toronto   
1        M5N  Central Toronto   
2        M4P  Central Toronto   
3        M5P  Central Toronto   
4        M4R  Central Toronto   

                                     Neighbourhood   Latitude  Longitude  
0                                    Lawrence Park  43.728020 -79.388790  
1                                         Roselawn  43.711695 -79.416936  
2                                 Davisville North  43.712751 -79.390197  
3  Forest Hill North & West, Forest Hill Road Park  43.696948 -79.411307  
4               North Toronto West,  Lawrence Park  43.715383 -79.405678

#### Now get the geolocation for Downtown Toronto

In [227]:
address = 'Downtown Toronto, ON'

geolocator = Nominatim(user_agent="dt_explorer")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print('The geograpical coordinate of Downtown Toronto are {}, {}.'.format(latitude, longitude))

The geograpical coordinate of Downtown Toronto are 43.6563221, -79.3809161.


In [228]:
# create map of Downtown Toronto using latitude and longitude values
map_downtown = folium.Map(location=[latitude, longitude], zoom_start=11)

# add markers to map
for lat, lng, label in zip(Downtown_data['Latitude'], Downtown_data['Longitude'], Downtown_data['Neighbourhood']):
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_downtown)  
    
map_downtown

In [229]:
Downtown_data.loc[0, 'Neighbourhood']

'Regent Park, Harbourfront'

#### Need to split out the multiple neighbourhoods in each row of the Downtown Toronto dataframe to enable more granular analysis

In [230]:
# Step 1 - We start with creating a new dataframe from the series with PostalCode as the index
Unique_neighbour = pd.DataFrame(Downtown_data.Neighbourhood.str.split(', ').tolist(), index=Downtown_data.PostalCode).stack() 

# Step 2 - We now want to get rid of the secondary index
# To do this, we will make EmployeeId as a column (it can't be an index since the values will be duplicate)
Unique_neighbour = Unique_neighbour.reset_index([0, 'PostalCode'])

# Step 3 - Set the column names for the new split data as we want them, otherwise Neighbourhood coolumn name shows as 0
Unique_neighbour.columns = ['PostalCode', 'Neighbourhood']

# Add the long and lat coords back into the dataframe
Unique_neighbour = pd.merge(Unique_neighbour, dfGeoloc, on='PostalCode')
Unique_neighbour.head()

PostalCode                  Neighbourhood   Latitude  Longitude
0        M5A                    Regent Park  43.654260 -79.360636
1        M5A                   Harbourfront  43.654260 -79.360636
2        M7A                   Queen's Park  43.662301 -79.389494
3        M7A  Ontario Provincial Government  43.662301 -79.389494
4        M5B                Garden District  43.657162 -79.378937

#### Recreating the Downtown Toronto map to show all neighbourhoods now they're all split out

In [231]:
# create map of Downtown Toronto using latitude and longitude values
map_downtown = folium.Map(location=[latitude, longitude], zoom_start=13)

# add markers to map
for lat, lng, label in zip(Unique_neighbour['Latitude'], Unique_neighbour['Longitude'], Unique_neighbour['Neighbourhood']):
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_downtown)  
    
map_downtown

#### Define foursquare credentials and version

In [236]:
CLIENT_ID = 'OCFTHA2C3X3WZMKCDJZAUJLJZTVQ30V0MWLD2IJY3B0EJB02' # your Foursquare ID
CLIENT_SECRET = 'VYHLSBR12TQEMWGFYEH3RAAXFEVJMAKW0ZBZPM3IZ3OA441U' # your Foursquare Secret
ACCESS_TOKEN = 'HNVUYIX22T342L3ZGFY5YAMP5GSFWUY1MCCW1GJOOM0L40MM' # your FourSquare Access Token
VERSION = '20200605'
LIMIT = 30
print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: OCFTHA2C3X3WZMKCDJZAUJLJZTVQ30V0MWLD2IJY3B0EJB02
CLIENT_SECRET:VYHLSBR12TQEMWGFYEH3RAAXFEVJMAKW0ZBZPM3IZ3OA441U


#### Get the locations name

In [237]:
Unique_neighbour.loc[0, 'Neighbourhood']

'Regent Park'

In [250]:
neighbourhood_latitude = Unique_neighbour.loc[0, 'Latitude'] # neighbourhood latitude value
neighbourhood_longitude = Unique_neighbour.loc[0, 'Longitude'] # neighbourhood longitude value

neighbourhood_name = Unique_neighbour.loc[0, 'Neighbourhood'] # neighbourhood name

print('Latitude and longitude values of {} are {}, {}.'.format(neighbourhood_name, 
                                                               neighbourhood_latitude, 
                                                               neighbourhood_longitude))

Latitude and longitude values of Regent Park are 43.6542599, -79.3606359.


In [244]:

LIMIT = 100 # limit of number of venues returned by Foursquare API

radius = 500 # define radius

# create URL
url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
    CLIENT_ID, CLIENT_SECRET, VERSION, neighbourhood_latitude, neighbourhood_longitude, radius, LIMIT)
url # display URL

'https://api.foursquare.com/v2/venues/explore?&client_id=OCFTHA2C3X3WZMKCDJZAUJLJZTVQ30V0MWLD2IJY3B0EJB02&client_secret=VYHLSBR12TQEMWGFYEH3RAAXFEVJMAKW0ZBZPM3IZ3OA441U&v=20200605&ll=43.6542599,-79.3606359&radius=500&limit=100'

In [254]:
results = requests.get(url).json()

In [281]:
# function that extracts the category of the venue
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

In [294]:
venues = results['response']['groups'][0]['items']

nearby_venues = json_normalize(venues) # flatten JSON

# filter columns
filtered_columns = ['venue.name', 'venue.categories', 'venue.location.lat', 'venue.location.lng']
nearby_venues = nearby_venues.loc[:, filtered_columns]
nearby_venues['venue.categories'] = nearby_venues.apply(get_category_type, axis=1)

nearby_venues.head()

AttributeError: module 'pandas._libs.reduction' has no attribute 'compute_reduction'

In [283]:
venues = results['response']['groups'][0]['items']
    
nearby_venues = json_normalize(venues) # flatten JSON

# filter columns
filtered_columns = ['venue.name', 'venue.categories', 'venue.location.lat', 'venue.location.lng']
nearby_venues = nearby_venues.loc[:, filtered_columns]

# filter the category for each row
nearby_venues['venue.categories'] = nearby_venues.apply(get_category_type, axis=1)

# clean columns
nearby_venues.columns = [col.split(".")[-1] for col in nearby_venues.columns]

nearby_venues.head()

AttributeError: module 'pandas._libs.reduction' has no attribute 'compute_reduction'

In [262]:
print('{} venues were returned by Foursquare.'.format(nearby_venues.shape[0]))
nearby_venues.head()

45 venues were returned by Foursquare.


venue.name                                   venue.categories  \
0        Roselle Desserts  [{'id': '4bf58dd8d48988d16a941735', 'name': 'B...   
1           Tandem Coffee  [{'id': '4bf58dd8d48988d1e0931735', 'name': 'C...   
2  Cooper Koo Family YMCA  [{'id': '52e81612bcbc57f1066b7a37', 'name': 'D...   
3     Body Blitz Spa East  [{'id': '4bf58dd8d48988d1ed941735', 'name': 'S...   
4          Impact Kitchen  [{'id': '4bf58dd8d48988d1c4941735', 'name': 'R...   

   venue.location.lat  venue.location.lng  
0           43.653447          -79.362017  
1           43.653559          -79.361809  
2           43.653249          -79.358008  
3           43.654735          -79.359874  
4           43.656369          -79.356980

In [266]:
def getNearbyVenues(names, latitudes, longitudes, radius=500):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighbourhood', 
                  'Neighbourhood Latitude', 
                  'Neighbourhood Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

In [267]:
DT_venues = getNearbyVenues(names=Unique_neighbour['Neighbourhood'],
                                   latitudes=Unique_neighbour['Latitude'],
                                   longitudes=Unique_neighbour['Longitude']
                                  )

Regent Park
Harbourfront
Queen's Park
Ontario Provincial Government
Garden District
Ryerson
St. James Town
Berczy Park
Central Bay Street
Christie
Richmond
Adelaide
King
Harbourfront East
Union Station
Toronto Islands
Toronto Dominion Centre
Design Exchange
Commerce Court
Victoria Hotel
University of Toronto
Harbord
Kensington Market
Chinatown
Grange Park
CN Tower
King and Spadina
Railway Lands
Harbourfront West
Bathurst Quay
South Niagara
Island airport
Rosedale
Stn A PO Boxes
St. James Town
Cabbagetown
First Canadian Place
Underground city
Church and Wellesley


In [268]:
print(DT_venues.shape)
DT_venues.head()

(2384, 7)


Neighbourhood  Neighbourhood Latitude  Neighbourhood Longitude  \
0   Regent Park                43.65426               -79.360636   
1   Regent Park                43.65426               -79.360636   
2   Regent Park                43.65426               -79.360636   
3   Regent Park                43.65426               -79.360636   
4   Regent Park                43.65426               -79.360636   

                    Venue  Venue Latitude  Venue Longitude  \
0        Roselle Desserts       43.653447       -79.362017   
1           Tandem Coffee       43.653559       -79.361809   
2  Cooper Koo Family YMCA       43.653249       -79.358008   
3     Body Blitz Spa East       43.654735       -79.359874   
4          Impact Kitchen       43.656369       -79.356980   

        Venue Category  
0               Bakery  
1          Coffee Shop  
2  Distribution Center  
3                  Spa  
4           Restaurant

In [273]:
DT_venues.groupby('Neighbourhood').count()

Neighbourhood Latitude  \
Neighbourhood                                           
Adelaide                                           96   
Bathurst Quay                                      16   
Berczy Park                                        56   
CN Tower                                           16   
Cabbagetown                                        44   
Central Bay Street                                 61   
Chinatown                                          61   
Christie                                           15   
Church and Wellesley                               77   
Commerce Court                                    100   
Design Exchange                                   100   
First Canadian Place                              100   
Garden District                                   100   
Grange Park                                        61   
Harbord                                            32   
Harbourfront                                       45   
Harbourfront East                                 100   
Harbourfront West                                  16   
Island airport                                     16   
Kensington Market                                  61   
King                                               96   
King and Spadina                                   16   
Ontario Provincial Government                      35   
Queen's Park                                       35   
Railway Lands                                      16   
Regent Park                                        45   
Richmond                                           96   
Rosedale                                            4   
Ryerson                                           100   
South Niagara                                      16   
St. James Town                                    123   
Stn A PO Boxes                                     97   
Toronto Dominion Centre                           100   
Toronto Islands                                   100   
Underground city                                  100   
Union Station                                     100   
University of Toronto                              32   
Victoria Hotel                                    100   

                               Neighbourhood Longitude  Venue  Venue Latitude  \
Neighbourhood                                                                   
Adelaide                                            96     96              96   
Bathurst Quay                                       16     16              16   
Berczy Park                                         56     56              56   
CN Tower                                            16     16              16   
Cabbagetown                                         44     44              44   
Central Bay Street                                  61     61              61   
Chinatown                                           61     61              61   
Christie                                            15     15              15   
Church and Wellesley                                77     77              77   
Commerce Court                                     100    100             100   
Design Exchange                                    100    100             100   
First Canadian Place                               100    100             100   
Garden District                                    100    100             100   
Grange Park                                         61     61              61   
Harbord                                             32     32              32   
Harbourfront                                        45     45              45   
Harbourfront East                                  100    100             100   
Harbourfront West                                   16     16              16   
Island airport                                      16     16              16   
Kensington Market                                

In [274]:
print('There are {} unique categories'.format(len(DT_venues['Venue Category'].unique())))

There are 207 unique categories


In [278]:
# one hot encoding
DT_onehot = pd.get_dummies(DT_venues[['Venue Category']], prefix="", prefix_sep="")

# add neighbourhood column back to dataframe
DT_onehot['Neighbourhood'] = DT_venues['Neighbourhood'] 

# move neighbourhood column to the first column
fixed_columns = [DT_onehot.columns[-1]] + list(DT_onehot.columns[:-1])
DT_onehot = DT_onehot[fixed_columns]

DT_onehot.head()

Neighbourhood  Adult Boutique  Airport  Airport Food Court  Airport Gate  \
0   Regent Park               0        0                   0             0   
1   Regent Park               0        0                   0             0   
2   Regent Park               0        0                   0             0   
3   Regent Park               0        0                   0             0   
4   Regent Park               0        0                   0             0   

   Airport Lounge  Airport Service  Airport Terminal  American Restaurant  \
0               0                0                 0                    0   
1               0                0                 0                    0   
2               0                0                 0                    0   
3               0                0                 0                    0   
4               0                0                 0                    0   

   Antique Shop  Aquarium  Art Gallery  Art Museum  Arts & Crafts Store  \
0             0         0            0           0                    0   
1             0         0            0           0                    0   
2             0         0            0           0                    0   
3             0         0            0           0                    0   
4             0         0            0           0                    0   

   Asian Restaurant  BBQ Joint  Baby Store  Bagel Shop  Bakery  Bank  Bar  \
0                 0          0           0           0       1     0    0   
1                 0          0           0           0       0     0    0   
2                 0          0           0           0       0     0    0   
3                 0          0           0           0       0     0    0   
4                 0          0           0           0       0     0    0   

   Baseball Stadium  Basketball Stadium  Beach  Bed & Breakfast  Beer Bar  \
0                 0                   0      0                0         0   
1                 0                   0      0                0         0   
2                 0                   0      0                0         0   
3                 0                   0      0                0         0   
4                 0                   0      0                0         0   

   Beer Store  Belgian Restaurant  Bistro  Boat or Ferry  Bookstore  Boutique  \
0           0                   0       0              0          0         0   
1           0                   0       0              0          0         0   
2           0                   0       0              0          0         0   
3           0                   0       0              0          0         0   
4           0                   0       0              0          0         0   

   Brazilian Restaurant  Breakfast Spot  Brewery  Bubble Tea Shop  Building  \
0                     0               0        0                0         0   
1                     0               0        0                0         0   
2                     0               0        0                0         0   
3                     0               0        0                0         0   
4                     0               0        0                0         0   

   Burger Joint  Burrito Place  Butcher  Café  Candy Store  \
0             0              0        0     0            0   
1             0              0        0     0            0   
2             0              0        0     0            0   
3             0              0        0     0            0   
4             0              0        0     0            0   

   Caribbean Restaurant  Cheese Shop  Chinese Restaurant  Chocolate Shop  \
0                     0            0                   0               0   
1                     0            0                   0               0   
2                     0            0                   0               0   
3                     0            0                  

In [279]:
DT_onehot.shape

(2384, 208)